src/Interceptor/RecordStartTime.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Amp\Http\Client\Interceptor;
+
+use Amp\CancellationToken;
+use Amp\Http\Client\ApplicationInterceptor;
+use Amp\Http\Client\DelegateHttpClient;
+use Amp\Http\Client\HarAttributes;
+use Amp\Http\Client\Request;
+use Amp\Promise;
+use function Amp\Internal\getCurrentTime;
+
+final class RecordStartTime implements ApplicationInterceptor
+{
+    public function request(Request $request, CancellationToken $cancellation, DelegateHttpClient $next): Promise
+    {
+        $request->setAttribute(HarAttributes::STARTED_DATE_TIME, new \DateTimeImmutable);
+        $request->setAttribute(HarAttributes::TIME_START, getCurrentTime());
```

Comment:
Should we just always record these, since many other Har attributes are being recorded in the connection pools.

Reply:
@trowski `HttpClientBuilder` always enables recording these, but they need to be the outer-most interceptor to have the correct time there.

Reply:
Ok, so this isn't really an optional interceptor then.

Reply:
It's optional as long as you don't want to write a log and don't use `HttpClientBuilder`.

Reply:
Yes, I meant not optional when using `HttpClientBuilder`, I didn't notice it was added there at first.